src/App.tsx

```diff
@@ -1,61 +1,126 @@
+import React, { useState } from 'react';
 import './App.scss';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { Users } from './types/Users';
+import { Todos } from './types/Todos';
+import { TodoList } from './components/TodoList';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+function getUserById(userId: number): Users | null {
+  return usersFromServer.find(person => person.id === userId) || null;
+}
+
+const todos: Todos[] = todosFromServer.map(todo => {
+  const user = getUserById(todo.userId);
+
+  return { ...todo, user };
+});
+
+function getNewTodoId(items: Todos[]) {
+  const maxId = Math.max(...items.map(item => item.id));
+
+  return maxId + 1;
+}
 
 export const App = () => {
+  const [todoses, setTodoses] = useState<Todos[]>(todos);
+  const [title, setTitle] = useState<string>('');
+  const [hasTitleError, setHasTitleError] = useState<boolean>(false);
+  const [userId, setUserId] = useState<number>(0);
+  const [hasUserError, setHasUserError] = useState<boolean>(false);
+
+  const onTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
```

Comment:
This function using inside this component, so, it would be better if you give a `handleTitleChange`
The logic is like this. In the parent component, you call the function with the `handle' prefix, in the child component you accept props with the name `on'. Then you will quickly distinguish which function and where you use it